nsxt/policy_common.go

```diff
@@ -310,13 +310,13 @@ func getSecurityPolicyAndGatewayRuleSchema(scopeRequired bool, isIds bool, nsxID
 	return ruleSchema
 }
 
-func getPolicyGatewayPolicySchema() map[string]*schema.Schema {
-	secPolicy := getPolicySecurityPolicySchema(false, true, true, true)
+func getPolicyGatewayPolicySchema(withDomain bool) map[string]*schema.Schema {
+	secPolicy := getPolicySecurityPolicySchema(false, true, true, withDomain)
 	// GW Policies don't support scope
 	delete(secPolicy, "scope")
 	secPolicy["category"].ValidateFunc = validation.StringInSlice(gatewayPolicyCategoryWritableValues, false)
 	// GW Policy rules require scope to be set
-	secPolicy["rule"] = getSecurityPolicyAndGatewayRulesSchema(true, false, true)
+	secPolicy["rule"] = getSecurityPolicyAndGatewayRulesSchema(withDomain, false, true)
```

Comment:
I thought the first parameter is for scope required.
if withDomain = False and then we have a VPC resource, does this make scope optional?

Reply:
I guess that I've made it so to make scope optional for VPC - but I'm not sure that it should be so.